TLM/TLM/UI/Helpers/CheckboxOption.cs

```diff
@@ -2,44 +2,137 @@ namespace TrafficManager.UI.Helpers {
     using ICities;
     using ColossalFramework.UI;
     using State;
+    using CSUtil.Commons;
+    using System.Collections.Generic;
+    using JetBrains.Annotations;
 
     public class CheckboxOption : SerializableUIOptionBase<bool, UICheckBox> {
-        public CheckboxOption(string fieldName) : base(fieldName) {
+        [CanBeNull]
+        private List<CheckboxOption> _requires;
+
+        public CheckboxOption(string fieldName, Options.PersistTo scope = Options.PersistTo.Savegame)
+        : base(fieldName, scope) {
             OnValueChanged = DefaultOnValueChanged;
         }
 
-        public event ICities.OnCheckChanged OnValueChanged;
+        /* Data */
 
+        public event OnCheckChanged OnValueChanged;
         public OnCheckChanged Handler {
             set => OnValueChanged += value;
         }
 
+        /// <summary>Optional list of checkboxes wich must be checked for this option to work.</summary>
+        /// <remarks>If specified, setting this option <c>true</c> will also set those other checkboxes <c>true</c>.</remarks>
+        [CanBeNull]
+        public List<CheckboxOption> Requires {
+            get => _requires;
+            set {
+                _requires = value;
+                Log.Info($"CheckboxOption.Requires: {nameof(Options)}.{FieldName} marked as requiring:");
+
+                foreach (var requirement in _requires) {
+                    Log.Info($"- {nameof(Options)}.{requirement.FieldName}");
+
+                    if (requirement.Dependents == null) {
+                        requirement.Dependents = new();
+                    }
+
+                    requirement.Dependents.Add(this);
+                }
+            }
+        }
+
+        /// <summary>Opional list of options that are depending on this option.</summary>
+        /// <remarks>Don't set directly; it's automatically managed by <see cref="Requires"/> property.</remarks>
+        [CanBeNull]
+        public List<CheckboxOption> Dependents { get; set; }
+
+        public override void Load(byte data) {
+            Log.Info($"CheckboxOption.Load: {data} -> {data != 0} -> {nameof(Options)}.{FieldName}");
```

Comment:
It will be toned down for later (many will be set to `Log._Debug()`).

Initially I want somewhat excessive logging, particularly while working on #1356.